odoo/addons/base/models/ir_http.py

```diff
@@ -396,6 +396,10 @@ def _binary_record_content(
         if not content:
             if model == 'ir.attachment':
                 content = record.raw
+            elif (field_def.related_field and
+                  field_def.related_field.name == 'raw' and
+                  field_def.related_field.model_name == 'ir.attachment'):
+                content = record[field]
```

Comment:
Is the `or b''` that is present on the `else` below possibly necessary in this case?

Reply:
Yes possibly, something down the line may expect a `bytes` type even if `b''` is falsy. I thought that the case was similar to the `content.raw` case but I suppose it's assumed that `attachment.raw` is always defined when we reach that point.